torch/csrc/api/include/torch/nn/modules/container/parameterdict.h

```diff
@@ -0,0 +1,135 @@
+#pragma once
+
+#include <torch/nn/cloneable.h>
+#include <torch/ordered_dict.h>
+#include <vector>
+
+namespace torch {
+namespace nn {
+
+class ParameterDictImpl : public Cloneable<ParameterDictImpl> {
+ public:
+  using Iterator = OrderedDict<std::string, Tensor>::Iterator;
+  using ConstIterator = OrderedDict<std::string, Tensor>::ConstIterator;
+
+  ParameterDictImpl() = default;
+
+  explicit ParameterDictImpl(
+      const torch::OrderedDict<std::string, torch::Tensor>& params) {
+    parameters_ = params;
```

Comment:
This is fine.
The reason you get compile error is that parameters_ is not a direct member variable of ParameterDictImpl, it can not be initialized using initialize_list in constructor. What I meant is that you can check whether 'Module' class has a constructor which takes 'params' as argument, so you can call 'Module' class's constructor to initialize it.